src/mds/MDSDaemon.cc

```diff
@@ -989,6 +989,7 @@ void MDSDaemon::handle_mds_map(MMDSMap *m)
             // has taken our ID, we don't want to keep restarting and
             // fighting them for the ID.
             suicide();
+            m->put();
```

Comment:
@gregsfortytwo  And for this one, because on any other exits(check https://github.com/ceph/ceph/pull/8408/commits/473371996d5b7c2402d1ef7d0901a5e350ba9673#diff-89e22a97695f436e6645ce25f7bfb985L914 and https://github.com/ceph/ceph/pull/8408/commits/473371996d5b7c2402d1ef7d0901a5e350ba9673#diff-89e22a97695f436e6645ce25f7bfb985R1035) are decreasing the message's reference by one.  
Although this one is not necessary because we are going to shutdown anyway, but it is a good habit to do the cleanup.